tests/phpunit/tests/functions/wpFastHash.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Test wp_fast_hash() function.
+ *
+ * @since 6.8.0
+ * @group functions
+ * @covers ::wp_fast_hash
+ */
+class Tests_Functions_wpFastHash extends WP_UnitTestCase {
+
+	/**
+	 * The schema of the wp_users table on wordpress.org has not been updated since the schema was changed in
+	 * [35638] for WordPress 4.4, which means the `user_activation_key` field remains at 60 characters length
+	 * instead of the expected 255. Although this is unlikely to affect others sites, let's accommodate for this
+	 * by ensuring that the length of this hash, which is used for password resets, does not exceed 60 characters.
+	 *
+	 * @ticket 21022
+	 */
+	public function test_wp_fast_hash_length_does_not_exceed_60_chars() {
```

Comment:
This test isn't quite correct for [the issue with un-upgraded tables](https://core.trac.wordpress.org/ticket/21022#comment:223) identified by @dd32.

The activation key is stored in the form `timestamp:$generic$<hash>`, for example on this branch it could be `1740260446:$generic$1SZb_XCetc,etc,etc`.

As the entire activation key field on legacy database formats is 60 characters long, we need to account for the timestamp prefix too.

We've got until November 2286 until the timestamp will increase in character length, so it's probably safe to assume that we only need to account for 10 digit timestamps for the time being.

Reply:
Excellent spot, you're completely correct. I'll add tests for this based on the ones you sent in Slack.